### ISSUE TITLE
elimination of lost particles

### ISSUE THREAD

**Issue by scpalbright** [blond-admin/BLonD#143]
Beam.eliminate_lost_particles is not working at all.

`
def eliminate_lost_particles(self):
        """Eliminate lost particles from the beam coordinate arrays
        """
        
        indexalive = np.where( self.id == 0 )[0]
        if len(indexalive) < self.n_macroparticles:
            self.dt = np.ascontiguousarray(self.beam.dt[indexalive])
            self.dE = np.ascontiguousarray(self.beam.dE[indexalive])
            self.n_macroparticles = len(self.beam.dt)
        else:
            #AllParticlesLost
            raise RuntimeError("ERROR in Beams: all particles lost and"+
" eliminated!") `

any call "self.beam" obviously fails because the beam object does not have a beam member, and indexalive is being set to those particles with id==0, which means we're keeping the particles that should be lost and vice versa.

I'm working on rewriting this so it works, but need to not break anyone else's code.  If correctly written this structure will remove the dt and dE for dead particles, but not their id so it will stop working after the first call.

Do any details of lost particles need to be stored, or can they be eliminated entirely?

**Comment by scpalbright** [comment]
Question for @kiliakis:  In the C++ functions for kick, drift, histogram etc would it be possible to include a tag to say which particles should/should not be included in the calculation?

**Comment by kiliakis** [comment]
Yes it is. I would rather add a separate C function for that to avoid any slowdown in case we don't want to use the tag and apply kick, drift, etc to all particles. Then in utis/bphysics_wrap.py we can write some code to choose which of the two functions to call based on the arguments passed.
